Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -342,6 +348,42 @@ void WebsiteDataStore::unregisterProcess(WebProcessProxy& process)
     m_processes.remove(process);
 }
 
+String WebsiteDataStore::migrateMediaKeysStorageIfNecessary(const String& directory)
+{
+    if (directory.isEmpty())
+        return emptyString();
+
+    static constexpr ASCIILiteral versionName = "v1"_s;
+    auto versionDirectory = FileSystem::pathByAppendingComponent(directory, versionName);
+    auto saltPath = FileSystem::pathByAppendingComponent(versionDirectory, "salt"_s);
+    m_mediaKeysStorageSalt = valueOrDefault(FileSystem::readOrMakeSalt(saltPath));
+
+    auto originDirectoryNames = FileSystem::listDirectory(directory);
+    // Migrate existing data to new version directory.
+    for (auto name : originDirectoryNames) {
```

Comment:
I don't see how we avoid doing this work every time. How does it detect that it has already migrated the media keys to the new format?

Reply:
It just tries to move old directories to new version directory every time. If there is no old origin directory, then no move is needed.